<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<!-- Configure the ezgripper module -->
	<arg name="ezgripper_module" default="dual_gen2" doc="Gripper module dictonary - {dual_gen1, dual_gen2, quad}" />

  <arg name="model" default="$(find ezgripper_driver)/urdf/$(arg ezgripper_module)/ezgripper_$(arg ezgripper_module)_standalone.urdf.xacro"/>
  <arg name="gui" default="true" />
  <arg name="rvizconfig" default="$(find ezgripper_driver)/rviz/urdf.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
  <param name="gui" value="$(arg gui)"/>

  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
		<remap from="joint_states" to="/ezgripper_$(arg ezgripper_module)/joint_states" />
	</node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
		<remap from="joint_states" to="/ezgripper_$(arg ezgripper_module)/joint_states" />
	</node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>
